indico/modules/events/registration/tasks.py

```diff
@@ -99,3 +100,14 @@ def delete_registrations():
         regform.is_purged = True
 
     db.session.commit()
+
+
+@celery.task(name='delete_previous_registration_file')
+def delete_previous_registration_file(reg_data, storage_backend, storage_file_id):
+    if storage_file_id is None:
+        return
+    if reg_data.storage_file_id == storage_file_id:
+        return
```

Comment:
I'd check both file id and backend. While extremely unlikely, two different backends could have the same file id for different files.